### PR TITLE
backup lambda improvement

### PR DESCRIPTION
**Pull request by aboulay**
- Add dynamic recuperation of EC2 informations (lambda).

### CODE REVIEW

**Comment by aboulay** [comment]
Waiting for right on the lambda IAM role. (#14219)

File: cmd/snapshot/main.go [hunk 1]
```diff
@@ -1,95 +1,145 @@
...
+
+	config := &ssh.ClientConfig{
+		User: os.Getenv("NODE_USER"),
+		Auth: []ssh.AuthMethod{
+			getPublicKeyFromS3(),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	instance_ip := getInstanceIP()
```
Comment (louptheron) [inline cmd/snapshot/main.go]: We are in camelCase :)

File: cmd/snapshot/main.go [hunk 2]
```diff
@@ -1,95 +1,145 @@
...
+			&ec2.Filter{
+				Name: aws.String("tag:BuildWith"),
+				Values: []*string{
+					aws.String("terraform"),
+				},
+			},
+			&ec2.Filter{
+				Name: aws.String("tag:aws:autoscaling:groupName"),
+				Values: []*string{
+					aws.String(" tzlink-mainnet"),
```
Comment (louptheron) [inline cmd/snapshot/main.go]: why a space before `tzlink-mainnet` ?

File: cmd/snapshot/main.go [hunk 3]
```diff
@@ -1,95 +1,145 @@
...
+	signer, err := ssh.ParsePrivateKey(buf.Bytes())
+	if err != nil {
+		panic(fmt.Sprintf("Unable to parse key %q at bucket %q, %v", S3LambdaKey, os.Getenv("S3_BUCKET"), err))
+	}
+
+	log.Print("got private key from S3 bucket.")
+	return ssh.PublicKeys(signer)
+}
+
+func getInstanceIP() *string {
```
Comment (louptheron) [inline cmd/snapshot/main.go:76]: Why returning a pointer ?

**Review by louptheron — COMMENTED** [review]
Some comments :)
Could you run `make fmt` before you push ? I saw some tabs are being replaced!

**Comment by aboulay** [comment]
Add tz_mode tag in the lambda in the filter

**Comment by aboulay** [comment]
Wait for dependancies on https://github.com/octo-technology/tezos-link/pull/44.
please merge this one before reviewing this

**Review by louptheron — APPROVED** [review]

**Pull request merged**